src/utils.c

```
@@ -196,10 +303,77 @@ int parse_cuda_visible_env() {
 }
 
 int map_cuda_visible_devices() {
+    int visible_count = 0;
+
     parse_cuda_visible_env();
+    if (CUDA_OVERRIDE_CALL(cuda_library_entry, cuDeviceGetCount,
```

Cppcheck (2.21.0)

[warning] If resource allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfResources)